zerver/tests/test_home.py

```diff
@@ -459,3 +460,111 @@ def test_subdomain_homepage(self):
             with patch('zerver.views.home.get_subdomain', return_value="subdomain"):
                 result = self._get_home_page()
             self._sanity_check(result)
+
+    def soft_deactivate_user(self, user):
+        # type: (str) -> UserProfile
+        long_term_idle_user = UserProfile.objects.get(
+            email=self.example_email(user))
+        long_term_idle_user.last_active_message_id = UserMessage.objects.filter(
+            user_profile=long_term_idle_user).order_by(
+            '-message__id')[0].message_id
+        long_term_idle_user.long_term_idle = True
+        long_term_idle_user.save(update_fields=[
+            'long_term_idle',
+            'last_active_message_id'])
+        return long_term_idle_user
+
+    def get_user_ums(self, user_profile):
+        # type: (UserProfile) -> List[UserMessage]
+        return list(UserMessage.objects.filter(
+            user_profile=user_profile).order_by('message__id'))
+
+    def send_stream_message(self, content, sender_name='iago',
+                            stream_name='Denmark', subject='foo'):
+        # type: (str, str, str, str) -> None
+        sender = self.example_email(sender_name)
+        self.send_message(sender, stream_name, Recipient.STREAM,
+                          content, subject)
+
+    def get_stream_unread_count(self, stream='Denmark', topic='foo'):
+        # type: (str, str) -> int
+        stream_narrow = self._get_home_page(stream=stream, topic=topic)
+        page_params = self._get_page_params(stream_narrow)
+        return page_params['unread_count']
+
+    def test_unread_count_user_soft_deactivation(self):
+        # type: () -> None
+        # In this test we make sure if a soft deactivated user had unread
+        # messages before deactivation they remain same way after activation.
+        self.login(self.example_email("hamlet"))
+        message = 'Test Message 1'
+        self.send_stream_message(message)
+        with queries_captured() as queries:
+            self.assertEqual(self.get_stream_unread_count(), 1)
+        query_count = len(queries)
+        user_msg_list = self.get_user_ums(self.example_user('hamlet'))
```

Comment:
I think this sort of code would read cleaner if you did `email = self.example_user("hamlet")` at the top rather than accessing it N times.